backend/fms_core/services/container.py

```diff
@@ -163,7 +168,11 @@ def create_container(barcode, kind, name=None, coordinates=None, container_paren
         if kind:
             if Container.objects.filter(barcode=barcode).exists():
                 errors.append(f"Container with barcode {barcode} already exists.")
+            elif name is None and Container.objects.filter(name=barcode):
```

Comment:
You need the ".exists()" or you will test if you have or not a queryset (which you most definitively have).